### PR TITLE
doc: improve depends prefix documentation

### PR DESCRIPTION
**Pull request by fanquake**
There seems to be some confusion about exactly how to use depends, when to pass a prefix to `./configure` etc (see #16367, #16654).

I've cherry-picked two of russ's commits out of #16367, as they are clear stand-alone improvements and we don't have to wait for #16367 to improve the depends documentation.

### CODE REVIEW

**Review by ryanofsky — APPROVED** [review]
ACK 0b1c2b0418ac39d6935bf3eb207915d33c9d983a. Maybe discount this review because I wrote these commits, but I agree they make sense in this standalone PR.

**Comment by ryanofsky** [comment]
Also, I think it would be a good idea that if a `depends/${host}/share/config.site` file exists, bitcoin `configure` script would just use `depends/$(host)` as the default `--prefix` value instead of `/usr/local`.

This way using depends would just require running 'make -C depends; ./configure; make` without needing to figure out the host triple or look at depends output.

**Comment by fanquake** [comment]
@ryanofsky Thanks. Probably out of scope to make those changes here, but I've opened #16692 to track.

**Comment by practicalswift** [comment]
Should it be clarified that `--prefix` is used not in its usual meaning of "when you run `./configure --prefix=/foo/bar`, you are indicating that the software should be installed under the `/foo/bar` directory".

That seems to be a recurring source of confusion.

Has it been discussed if we could use something else than `--prefix` for the "where to find dependencies"-meaning we're talking about here?

File: depends/README.md [hunk 1]
```diff
@@ -12,14 +12,17 @@ For example:
...
+order for it to pick up libraries, tools, and settings from the depends build,
+you must point it at the appropriate `--prefix` directory generated by the
+build. In the above example, a prefix dir named x86_64-w64-mingw32 will be
 created. To use it for Bitcoin:
 
     ./configure --prefix=`pwd`/depends/x86_64-w64-mingw32
 
 Common `host-platform-triplets` for cross compilation are:
 
+- `x86_64-pc-linux-gnu` for x86 Linux
```
Comment (hebasto) [inline depends/README.md]: Could the `i686-pc-linux-gnu` triplet be added?

**Review by hebasto — COMMENTED** [review]
Concept ACK

**Review by hebasto — APPROVED** [review]
ACK 2483266c591f7b2e62df68ee2d13740a706415ec, I have reviewed the code and it looks OK, I agree it can be merged.

**Comment by Sjors** [comment]
utACK 2483266

**Review by jonasschnelli — APPROVED** [review]
ACK 2483266c591f7b2e62df68ee2d13740a706415ec

File: depends/packages.md [hunk 2]
```diff
@@ -181,3 +181,18 @@ For us, it's much easier to just link a static `libsecondary` into a shared
 static or dynamic `libseconday`, that's not our concern. With a static
 `libseconday`, when we need to link `libprimary` into our executable, there's no
 dependency chain to worry about as `libprimary` has all the symbols.
+
+## Build targets:
+
+To build an individual package (useful for debugging), following build targets are available.
```
Comment (jonatack) [inline depends/packages.md]: nit: missing "the" before "following"

**Review by jonatack — COMMENTED** [review]
ACK. After my questions on the #bitcoin-builds irc channel, like you I was thinking this info would be useful to add here.

File: depends/packages.md (hunk 2 above)
Comment (jonatack) [reply depends/packages.md]: Will retouch it on the side in another PR.

**Pull request merged**